frontend/src/lib/utils.tsx

```diff
@@ -110,6 +110,9 @@ export function deleteWithUndo({ undo = false, ...props }: Record<string, any>):
             onClick: () => {
                 deleteWithUndo({ undo: true, ...props })
             },
+            onClose: () => {
+                api.delete(`api/${props.endpoint}/${props.object.id}`)
```

Comment:
Not 100% sure about this approach. What if this request fails? We already told the user it was successful. This could cause a quite a bit of headaches. That being said, handling these cases correctly on the server side might be too complicated to be worth it, so I wouldn't be opposed to merging it as it is. Though I think a better approach (and sort of middle ground) would be to ask for confirmation before deleting and then really executing the request (I think having both the delete to undo and confirm to delete UX for a transition period is fine).

Reply:
Replied below

Reply:
Yeah I'm also a fan of confirmation + real deletion, but that'd need a larger discussion.

And indeed, if the request fails then it won't be properly deleted. However, this PR should be a no side effects thing, since by default we're not fully deleting these anyway (we just often hide deleted=false objects). We also don't expect/handle failures from the PATCH request.